pepper-apis/src/main/java/org/broadinstitute/ddp/service/AgeUpService.java

```diff
@@ -88,39 +92,85 @@ public int runAgeUpCheck(Handle handle, PexInterpreter interpreter, GovernancePo
             boolean agedUp = rule.hasReachedAgeOfMajority(candidate.getBirthDate(), today);
             boolean shouldPrepForAgeUp = rule.hasReachedAgeOfMajorityPrep(candidate.getBirthDate(), today).orElse(false);
 
+            if (agedUp && candidate.getStatus() != EnrollmentStatusType.CONSENT_SUSPENDED) {
+                handle.savepoint(checkpoint("status", studyGuid, userGuid));
+                try {
+                    jdbiEnrollment.suspendUserStudyConsent(candidate.getParticipantUserId(), policy.getStudyId());
+                    LOG.info("Candidate {} in study {} has reached age-of-majority, consent status is suspended", userGuid, studyGuid);
+                    handle.release(checkpoint("status", studyGuid, userGuid));
+                } catch (Exception e) {
+                    LOG.error("Candidate {} in study {} has reached age-of-majority but could not suspend consent status, skipping",
+                            userGuid, studyGuid, e);
+                    handle.rollbackToSavepoint(checkpoint("status", studyGuid, userGuid));
+                    continue;
+                }
+            }
+
+            if (shouldPrepForAgeUp && !candidate.hasInitiatedPrep()) {
+                LOG.info("Candidate {} in study {} has reached preparation for age-of-majority, processing age-up preparation events",
+                        userGuid, studyGuid);
+                handle.savepoint(checkpoint("prep", studyGuid, userGuid));
+                try {
+                    EventSignal signal = new EventSignal(
+                            candidate.getParticipantUserId(),
+                            candidate.getParticipantUserId(),
+                            candidate.getParticipantUserGuid(),
+                            policy.getStudyId(),
+                            EventTriggerType.REACHED_AOM_PREP);
+                    EventService.getInstance().processAllActionsForEventSignal(handle, signal);
+                    preppedCandidateIds.add(candidate.getId());
+                    handle.release(checkpoint("prep", studyGuid, userGuid));
```

Comment:
Might be more robust to have a `finally` do the `release` here and above.  I'm not sure if there would need to be additional error handling if the savepoint is gone after rolling back.